live-ingester/src/test/java/org/alfresco/hxi_connector/live_ingester/adapters/auth/HxAuthenticationClientIntegrationTest.java

```diff
@@ -0,0 +1,160 @@
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Alfresco. If not, see <http://www.gnu.org/licenses/>.
+ * #L%
+ */
+package org.alfresco.hxi_connector.live_ingester.adapters.auth;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+
+import static com.github.tomakehurst.wiremock.client.WireMock.aResponse;
+import static com.github.tomakehurst.wiremock.client.WireMock.badRequest;
+import static com.github.tomakehurst.wiremock.client.WireMock.givenThat;
+import static com.github.tomakehurst.wiremock.client.WireMock.post;
+import static com.github.tomakehurst.wiremock.client.WireMock.postRequestedFor;
+import static com.github.tomakehurst.wiremock.client.WireMock.serverError;
+import static com.github.tomakehurst.wiremock.client.WireMock.urlPathEqualTo;
+import static org.apache.hc.core5.http.ContentType.APPLICATION_FORM_URLENCODED;
+import static org.apache.http.HttpHeaders.HOST;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.catchThrowable;
+import static org.mockito.BDDMockito.then;
+import static org.mockito.Mockito.times;
+
+import com.github.tomakehurst.wiremock.client.WireMock;
+import com.github.tomakehurst.wiremock.matching.EqualToPattern;
+import org.apache.camel.Exchange;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.springframework.boot.autoconfigure.EnableAutoConfiguration;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.mock.mockito.SpyBean;
+import org.springframework.retry.annotation.EnableRetry;
+import org.springframework.security.oauth2.client.registration.ClientRegistration;
+import org.springframework.test.context.DynamicPropertyRegistry;
+import org.springframework.test.context.DynamicPropertySource;
+import org.testcontainers.junit.jupiter.Container;
+import org.testcontainers.junit.jupiter.Testcontainers;
+import org.testcontainers.utility.DockerImageName;
+import org.wiremock.integrations.testcontainers.WireMockContainer;
+
+import org.alfresco.hxi_connector.live_ingester.adapters.config.IntegrationProperties;
+import org.alfresco.hxi_connector.live_ingester.domain.exception.EndpointClientErrorException;
+import org.alfresco.hxi_connector.live_ingester.domain.exception.EndpointServerErrorException;
+import org.alfresco.hxi_connector.live_ingester.util.AuthUtils;
+import org.alfresco.hxi_connector.live_ingester.util.DockerTags;
+
+@SpringBootTest(classes = {
+        IntegrationProperties.class,
+        HxAuthenticationClient.class},
+        properties = "logging.level.org.alfresco=DEBUG")
+@EnableAutoConfiguration
+@EnableRetry
+@Testcontainers
+class HxAuthenticationClientIntegrationTest
+{
+    private static final String WIREMOCK_IMAGE = "wiremock/wiremock";
+    private static final String WIREMOCK_TAG = DockerTags.getOrDefault("wiremock.tag", "3.3.1");
```

Comment:
At some point we should revisit this, because these defaults are going to get out of date and they're all over our tests. No need to change this PR for it though.

Reply:
Good point. Versions from `pom.xml` should be picked up, but we could place all defaults in `static final`s and gather them in one place 🤔  maybe in `DockerTags` class. This should be done in separate PR